pkg/core/basic_chain_test.go

```diff
@@ -596,3 +625,1634 @@ func TestBlockchain_StartFromExistingDB(t *testing.T) {
+// native contract is disabled. It's easy to forget about config while adding new
+// native contract.
+func TestConfigNativeUpdateHistory(t *testing.T) {
+	var prefixPath = filepath.Join("..", "..", "config")
+	check := func(t *testing.T, cfgFileSuffix interface{}) {
+		cfgPath := filepath.Join(prefixPath, fmt.Sprintf("protocol.%s.yml", cfgFileSuffix))
+		cfg, err := config.LoadFile(cfgPath)
+		require.NoError(t, err, fmt.Errorf("failed to load %s", cfgPath))
+		natives := native.NewContracts(cfg.ProtocolConfiguration)
+		assert.Equal(t, len(natives.Contracts),
+			len(cfg.ProtocolConfiguration.NativeUpdateHistories),
+			fmt.Errorf("protocol configuration file %s: extra or missing NativeUpdateHistory in NativeActivations section", cfgPath))
+		for _, c := range natives.Contracts {
+			assert.NotNil(t, cfg.ProtocolConfiguration.NativeUpdateHistories[c.Metadata().Name],
+				fmt.Errorf("protocol configuration file %s: configuration for %s native contract is missing in NativeActivations section; "+
+					"edit the test if the contract should be disabled", cfgPath, c.Metadata().Name))
+		}
+	}
+	testCases := []interface{}{
+		netmode.MainNet,
+		netmode.PrivNet,
+		netmode.TestNet,
+		netmode.UnitTestNet,
+		"privnet.docker.one",
+		"privnet.docker.two",
+		"privnet.docker.three",
+		"privnet.docker.four",
+		"privnet.docker.single",
+		"unit_testnet.single",
+	}
+	for _, tc := range testCases {
+		check(t, tc)
+	}
+}
+
+func TestBlockchain_VerifyTx(t *testing.T) {
+	bc, validators, committee := chain.NewMultiWithCustomConfig(t, func(c *config.ProtocolConfiguration) {
+		c.P2PSigExtensions = true
+		c.ReservedAttributes = true
+	})
+	e := neotest.NewExecutor(t, bc, validators, committee)
+
+	accs := make([]*wallet.Account, 5)
+	for i := range accs {
+		var err error
+		accs[i], err = wallet.NewAccount()
+		require.NoError(t, err)
+	}
+
+	notaryServiceFeePerKey := bc.GetNotaryServiceFeePerKey()
+
+	oracleAcc := accs[2]
+	oraclePubs := keys.PublicKeys{oracleAcc.PrivateKey().PublicKey()}
+	require.NoError(t, oracleAcc.ConvertMultisig(1, oraclePubs))
+
+	neoHash := e.NativeHash(t, nativenames.Neo)
+	gasHash := e.NativeHash(t, nativenames.Gas)
+	policyHash := e.NativeHash(t, nativenames.Policy)
+	designateHash := e.NativeHash(t, nativenames.Designation)
+	notaryHash := e.NativeHash(t, nativenames.Notary)
+
+	neoValidatorsInvoker := e.ValidatorInvoker(neoHash)
+	gasValidatorsInvoker := e.ValidatorInvoker(gasHash)
+	policySuperInvoker := e.NewInvoker(policyHash, validators, committee)
+	designateSuperInvoker := e.NewInvoker(designateHash, validators, committee)
+	neoOwner := validators.ScriptHash()
+
+	neoAmount := int64(1_000_000)
+	gasAmount := int64(1_000_000_000)
+	for _, a := range accs {
+		neoValidatorsInvoker.Invoke(t, true, "transfer", neoOwner, a.Contract.ScriptHash(), neoAmount, nil)
+		gasValidatorsInvoker.Invoke(t, true, "transfer", neoOwner, a.Contract.ScriptHash(), gasAmount, nil)
```

Comment:
It's a tx and a block for every transfer (with all multisignature overhead). Can we do it more efficiently?

Reply:
Sure, fixed.